torch/_inductor/inductor_prims.py

```diff
@@ -93,3 +93,36 @@ def eager_force_stride(input_tensor: Tensor, stride) -> Tensor:
     lambda a, b, c: (a * b) + c,
     doc="Fused multiply add: fma(a, b, c) -> (a * b) + c without rounding after the multiplication",
 )
+
+
+def _low_memory_max_pool2d_with_offsets_aten(
+    self, kernel_size, stride, padding, dilation, ceil_mode, *, offset_dtype
+):
+    vals, indices = torch.ops.aten.max_pool2d_with_indices(
+        self, kernel_size, stride, padding, dilation, ceil_mode
+    )
+    return vals, indices.to(offset_dtype)
+
+
+def _low_memory_max_pool2d_with_offsets_meta(
+    self, kernel_size, stride, padding, dilation, ceil_mode, *, offset_dtype
+):
+    vals, indices = torch.ops.aten.max_pool2d_with_indices(
+        self, kernel_size, stride, padding, dilation, ceil_mode
+    )
+    return _prims.TensorMeta(vals), _prims.TensorMeta(indices.to(offset_dtype))
+
+
+_low_memory_max_pool2d_with_offsets = _prims._make_prim(
+    schema="_low_memory_max_pool2d_with_offsets(Tensor self, SymInt[2] kernel_size, SymInt[2] stride,  SymInt[2] padding, SymInt[2] dilation, bool ceil_mode, *, ScalarType offset_dtype) -> (Tensor, Tensor)",
+    return_type=(_prims.RETURN_TYPE.NEW, _prims.RETURN_TYPE.NEW),
+    meta=_low_memory_max_pool2d_with_offsets_meta,
+    impl_aten=_low_memory_max_pool2d_with_offsets_aten,
+    doc="Instead of returning indices, returns indices offsets.",
+)
+
+_low_memory_max_pool2d_offsets_to_indices = make_prim(
+    "_low_memory_max_pool2d_offsets_to_indices(Tensor self, SymInt kernel_h, SymInt input_w, SymInt[2] stride, SymInt[2] padding) -> Tensor",
+    lambda self, *args: self.to(torch.int64),
```

Comment:
Do we want to change the dtype here dynamically ? typically we dont need int64 right ?

Reply:
I don't think that would be sound in general. It should be internal to the `offset_to_index->backward` fusion. There are tests which fail if output dtypes do not match eager.